### PR TITLE
add thanos-operator

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] REQUIRED - The package is added to `packages.txt`

#### For security-related PRs

- [ ] The security fix is recorded in `advisories` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

**Review by jdolitsky — DISMISSED** [review]

**Review by ajayk — CHANGES_REQUESTED** [review]
Please add the CVE fixes for the package

File: thanos-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/banzaicloud/thanos-operator
+      tag: ${{package.version}}
+      expected-commit: 7b87b4f89dee6454ae46e9579b9a98d6d7a5e550
+
+  - runs: |
+      go build -o bin/manager main.go
```
Comment (ajayk) [inline thanos-operator.yaml:24]:
```Suggestion 
      # Address  CVE-2022-21698  GHSA-69ch-w2m2-3vjp  GHSA-69cg-p879-7622
      go get github.com/prometheus/client_golang@v1.11.1
      go get golang.org/x/text@v0.3.8
      go get golang.org/x/net@v0.7.0
      
      go build -o bin/manager main.go
```

Comment (developer-guy) [reply thanos-operator.yaml:24]: thanks so much for this! I was also planning to do the same thing, but you beat me!

File: thanos-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  - name: "thanos-operator-compat"
+    description: "Compatibility package to place binaries in the location expected by upstream helm charts"
+    pipeline:
+      - runs: |
+          # The helm chart expects the thanos-operator binaries to be in / instead of /usr/bin
+          mkdir -p "${{targets.subpkgdir}}"
+          ln -sf /usr/bin/manager ${{targets.subpkgdir}}/manager
+      - uses: strip
+
+update:
```
Comment (rawlingsj) [inline thanos-operator.yaml:47]: looks like the auto update CI check us failing as there are SDK releases on this repo.  You can ignore these and have the CI check go green using https://github.com/chainguard-dev/melange/blob/main/docs/UPDATE.md#ignore-versions

File: thanos-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      - go
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/banzaicloud/thanos-operator
+      tag: ${{package.version}}
+      expected-commit: 7b87b4f89dee6454ae46e9579b9a98d6d7a5e550
+
+  - runs: |
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply thanos-operator.yaml:23]: cool, didn't know this pipeline supports deps option to fix CVEs, great!

File: thanos-operator.yaml (hunk 2 above)
Comment (developer-guy) [reply thanos-operator.yaml:47]: what should be the value of the `ignore-regex-patterns`? I couldn't understand what it should be :(

**Review by ajayk — APPROVED** [review]

**Review by ajayk — APPROVED** [review]

**Pull request merged**